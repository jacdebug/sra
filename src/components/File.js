import React from 'react';
import filesize from 'filesize';
import dateformat from 'dateformat';
import styled from 'styled-components';

const File = ({ file }) =>
  <FileListItem>
    <FileName>
      {file.name}
    </FileName>

    <div>
      {file.desc}
    </div>

    <FileMeta>
      <span>#{file.id}</span> |
      <span>{filesize(file.size)}</span> |
      <span>{file.ext}</span> |
      <span>{dateformat(file.created)}</span>
    </FileMeta>
  </FileListItem>;

const FileListItem = styled.div`
  color: #444;
  background: #f9f2f0;
  background: linear-gradient(to bottom, #fff9e7 0%, #f9f2f0 20%, #ffffff 100%);
  border-radius: 3px;
  border: 1px solid #e4d3d5;
  padding: 10px;
  margin: 5px 0;
  text-align: left;
`;

const FileName = styled.div`
  font-weight: bold;
  font-size: 14px;
`;

const FileMeta = styled.div`margin: 10px 0 0;`;

export default File;
