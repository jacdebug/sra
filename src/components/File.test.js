import React from 'react';
import { shallow } from 'enzyme';
import filesize from 'filesize';
import dateformat from 'dateformat';
import renderer from 'react-test-renderer';
import File from './File';

const fileData = {
  id: 0,
  name: 'checking_account.m2a',
  ext: 'm2a',
  desc: 'Minima dolor et sequi dicta vitae possimus molestiae laborum quis.',
  created: '2017-01-22T11:45:28.533Z',
  size: 2042211209,
};

const fileDataKeys = Object.keys(fileData);

it('renders without crashing', () => {
  shallow(<File file={fileData} />);
});

it('display all file informtion', () => {
  const filesShallowRender = shallow(<File file={fileData} />);
  const renderHtml = filesShallowRender.html();
  for (let key of fileDataKeys) {
    let testString = fileData[key];
    if (key === 'created') testString = dateformat(testString);
    if (key === 'size') testString = filesize(testString);
    expect(renderHtml.indexOf(testString) > -1).toBe(true);
  }
});

it('renders a snapshot', () => {
  const tree = renderer.create(<File file={fileData} />).toJSON();
  expect(tree).toMatchSnapshot();
});
