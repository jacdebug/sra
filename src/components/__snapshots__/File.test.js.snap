// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div
  className="sc-bdVaJa bmRMAg"
>
  <div
    className="sc-bwzfXH yQIlD"
  >
    checking_account.m2a
  </div>
  <div>
    Minima dolor et sequi dicta vitae possimus molestiae laborum quis.
  </div>
  <div
    className="sc-htpNat cIzfuY"
  >
    <span>
      #
      0
    </span>
     |
    <span>
      1.9 GB
    </span>
     |
    <span>
      m2a
    </span>
     |
    <span>
      Sun Jan 22 2017 11:45:28
    </span>
  </div>
</div>
`;
